internal/fwserver/server_importresourcestate_test.go

```diff
@@ -280,7 +289,53 @@ func TestServerImportResourceState(t *testing.T) {
 				},
 			},
 		},
-		"response-importedresources-deferral": {
+		"response-importedresources-deferral-automatic": {
+			server: &fwserver.Server{
+				Provider: &testprovider.Provider{
+					SchemaMethod: func(_ context.Context, _ provider.SchemaRequest, resp *provider.SchemaResponse) {},
+					ConfigureMethod: func(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
+						resp.Deferred = &provider.Deferred{Reason: provider.DeferredReasonProviderConfigUnknown}
+					},
+				},
+			},
+			configureProviderReq: &provider.ConfigureRequest{
+				ClientCapabilities: provider.ConfigureProviderClientCapabilities{
+					DeferralAllowed: true,
+				},
+			},
+			request: &fwserver.ImportResourceStateRequest{
+				EmptyState: *testEmptyState,
+				ID:         "test-id",
+				Resource: &testprovider.ResourceWithImportState{
+					Resource: &testprovider.Resource{},
+					ImportStateMethod: func(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+						if req.ID != "test-id" {
+							resp.Diagnostics.AddError("unexpected req.ID value: %s", req.ID)
+						}
+
+						resp.Deferred = &resource.Deferred{
+							Reason: resource.DeferredReasonAbsentPrereq,
+						}
+
+						resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+
```

Comment:
This logic shouldn't be executed I believe. Maybe we instead just return a diagnostic in this logic to ensure it fails the test if called?

https://github.com/hashicorp/terraform-plugin-sdk/blob/4e590aed07ae4b94ea765b07a35ae3c14fc4284b/helper/schema/grpc_provider_test.go#L5091-L5095